code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -104,8 +103,19 @@
 			if(M.satiety > -200)
 				M.satiety -= junkiness
 			playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), 1)
+			var/fraction = min(bitesize / reagents.total_volume, 1)
```

Comment:
You dropped the division by zero check. Also same as before isolate in carbon proc.